rotkehlchen/chain/evm/decoding/paraswap/decoder.py

```diff
@@ -101,6 +99,9 @@ def _decode_swap(
             return DEFAULT_DECODING_OUTPUT
 
         if partial_refund_event is not None:  # if some in_asset is returned back
+            if in_event.asset == out_event.asset:  # check the similarity of asset with out_event
+                # due to order of events it can happen that this two get mislabeled and need to be swapped  # noqa: E501
```

Comment:
Can you also maybe describe in a tiny bit more detail like you did to @yabirgb 's question in the comment here? A potential reader would need to understand when something may happen and why a check is in place.

```suggestion
                # due to order of events it can happen that these two get mislabeled and need to be swapped  # noqa: E501
```